libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -62,14 +71,21 @@ export class MessageRepository extends BaseRepository<MessageDBModel, MessageEnt
       requestQuery.read = { $in: [true, false] };
     }
 
+    if (query.payload) {
+      requestQuery = {
+        ...requestQuery,
+        ...getFlatObject({ payload: query.payload }),
```

Comment:
what if `query.payload` is an empty object, `{ }`. Are we still covered with errors/will not throw errors? Also in the conversion of the payload in other places ?

Reply:
if the `query.payload` is `{}` then the `getFlatObject({ payload: {} })` will return an empty object `{}` ;) 
So basically it won't return any keys flattened if the value is an empty object, including nested objects.